src/transformers/testing_utils.py

```diff
@@ -1870,9 +1870,18 @@ def preprocess_string(string, skip_cuda_tests):
         ):
             is_cuda_found = True
             break
+
     modified_string = ""
     if not is_cuda_found:
         modified_string = "".join(codeblocks)
+
+        if ">>>" in modified_string:
```

Comment:
only add `>>> import gc; gc.collect() # doctest: +IGNORE_RESULT` when the docstring containing `>>>`. Otherwsie too many doctest test cases and running time explodes.